extensions/podman/src/podman-install.ts

```diff
@@ -0,0 +1,202 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { downloadFile, fetchJson, isMac, isWindows } from './util';
+import * as fs from 'node:fs';
+import * as os from 'node:os';
+import * as path from 'node:path';
+import { spawn } from 'node:child_process';
+import * as extensionApi from '@tmpwip/extension-api';
+import { rcompare } from 'semver';
+
+import * as podmanTool from './podman.json';
+
+interface PodmanAsset {
+  name: string;
+  browser_download_url: string;
+}
+
+interface PodmanRelease {
+  body: string;
+  name: string;
+  assets: PodmanAsset[];
+}
+
+export async function installBundledPodman(): Promise<boolean> {
+  if (isWindows) {
+    return installWinBundledPodman();
+  } else if (isMac) {
+    return installMacBundledPodman();
+  } else {
+    //TODO: support podman installation on Linux OS
+    return false;
+  }
+}
+
+export async function installPodman(release: PodmanRelease): Promise<boolean> {
+  if (isWindows) {
+    return installWinPodman(release);
+  } else if (isMac) {
+    // return installMacPodman(release);
+  } else {
+    //TODO: support podman installation on Linux OS
+    return false;
+  }
+}
+
+export function getBundledPodmanVersion(): string {
+  return podmanTool.version;
+}
+
+export async function fetchLatestPodmanVersion(): Promise<PodmanRelease | undefined> {
+  const releases = await fetchLatestReleases();
+  return getLatestV4Release(releases);
+}
+
+function getAssetsFolder(): string {
+  const dirName = __dirname;
+  // this is temporary, as process.resourcesPath is not available in preload scripts
+  if (dirName.includes('app.asar')) {
+    const resourcesPath = dirName.substring(0, dirName.indexOf('app.asar'));
+    return path.resolve(resourcesPath, 'extensions', 'podman', 'builtin', 'podman.cdix', 'assets');
+  } else {
+    return path.resolve(__dirname, '..', 'assets');
+  }
+}
+
+async function installWinBundledPodman(): Promise<boolean> {
+  return extensionApi.window.withProgress(async progress => {
+    progress.report({ increment: 5 });
+    const msiPath = path.resolve(getAssetsFolder(), `podman-v${podmanTool.version}.msi`);
+    try {
+      if (fs.existsSync(msiPath)) {
+        await executeInstaller('msiexec', ['/i', msiPath, '/qb', '/norestart']);
+        progress.report({ increment: 80 });
+        await extensionApi.window.showDialog('info', 'Podman', 'Podman is successfully installed.', 'OK');
```

Comment:
is this modal? Not sure if this is helpful as we can assume the installation happens and the end result should be a working `machine init`. I think an error during install is more important.